CMakeLists.txt

```diff
@@ -117,6 +117,9 @@ if (CUB_ENABLE_TESTING OR CUB_ENABLE_EXAMPLES)
 endif()
 
 if (CUB_ENABLE_TESTING)
+  include(cmake/CPM.cmake)
+  CPMAddPackage("gh:catchorg/Catch2@2.13.7")
```

Comment:
Since we're adding this fresh, we may want to look at using v3 right away. In cuCo I found the compile time can be quite bad, and that's precisely what v3 is meant to address. 

https://github.com/catchorg/Catch2/blob/devel/docs/migrate-v2-to-v3.md#top